seatunnel-api/src/main/java/org/apache/seatunnel/api/tracing/MDCStream.java

```diff
@@ -53,62 +53,69 @@ public MDCStream(MDCContext context, Stream<T> delegate) {
 
     @Override
     public Stream<T> filter(Predicate<? super T> predicate) {
-        return new MDCStream<>(context, delegate.filter(new MDCPredicate<>(context, predicate)));
+        return new MDCStream<>(
+                context,
+                delegate.filter(new MDCPredicate<>(() -> MDCContext.of(context), predicate)));
```

Comment:
Because element iteration caused repeated execution, it was changed to lazy initialization.